module-2/backend/Liam_barstad/week_two.txt

```diff
@@ -0,0 +1,104 @@
+## Week Two - Module 2 Recap
+
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week (which is a TON - YOU are a web developer!!!).
+
+Note: When you're done, submit a PR.
+
+1. At a high level, what is ActiveRecord? What does it do/allow you to do?
+
+  ActiveRecord streamlines relationships between ruby objects, and, by extension, database elements. It also
+  provides the most quick, concise, and readable methods for handling basic functions dealing with datasets
+  and relationships.
+
+2. Assume you have the following model:
+
+```ruby
+class Team << ActiveRecord::Base
+end
+```
+
+What are some methods you can call on `Team`? If these methods aren't defined in the class, how do you have access to them?
+
+  - Team.all
+  - Team.create(<< parameters >>)
+  - Team.where(:<< instance variable >> = << value >>)
+  - Team.maximum(:<< instance variable >>)
+  - Team.pluck(:<< instance variable >>)
+We have access to these methods by default because they are defined in the ActiveRecord parent class, the
+requirements for which are initialized when a new rails application is created through the terminal (required
+in Gemfile etc. by default).
+
+3. Assume that in your database, a team has the following attributes: "id", "name", owner_id". How would you find the name of a team with an id of 4? Assuming your class only included the code from question 2, how could you find the owner of the same team?
+
+  team = Team.find(4)
+  owner_id = team.owner_id
```

Comment:
This will find the `owner_id`, but in order to find the owner we need one more step:

```ruby
Owner.find(owner_id)
```